### ISSUE TITLE
Sox issue

### ISSUE THREAD

**Issue by Maxopoly** [lyrebird-voice-changer/lyrebird#34]
Followed all installation instructions and ended up with
```
sox FAIL formats: can't open output file `Lyrebird-Output': can not open audio device: No such entity
Loaded null output sink (0), and remap sink (0)
sox WARN pitch: pitch clipped 9124 samples; decrease volume?
sox WARN rate: rate clipped 16448 samples; decrease volume?
Loaded null output sink (0), and remap sink (0)
sox WARN pitch: pitch clipped 3156 samples; decrease volume?
sox WARN rate: rate clipped 5610 samples; decrease volume?
sox FAIL formats: can't open output file `Lyrebird-Output': can not open audio device: No such entity
sox FAIL formats: can't open output file `Lyrebird-Output': can not open audio device: No such entity
```
Using python 3.7.9 on Ubuntu 18.04

**Comment by chxrlt** [comment]
did you install all the dependencies listed in the README?

**Comment by chxrlt** [comment]
more specifically, i believe ubuntu doesn't bundle `libsox-fmt-pulse` so you will need to install it manually using `sudo apt install libsox-fmt-pulse`. give that a try, and let me know if that solved your issue.

\- charlotte

**Comment by Maxopoly** [comment]
Yep, I had already installed that, did everything listed in the README, including the symlinks

**Comment by chxrlt** [comment]
hm not sure then, im tagging @harrego maybe he will be of more use

**Comment by Maxopoly** [comment]
Not sure if related or a separate issue, but I was also getting 
```
Module module-null-sink not loaded.
Module module-remap-source not loaded.
```
unless I manually loaded them in beforehand via
```
pactl load-module module-remap-source
pactl load-module module-null-sink
```

**Comment by Phewda** [comment]
I have the same issue.
The program worked correctly for a few seconds then I was not able to fix the problem even reinstalling the required libs again and again.

**Comment by thepenguinthatwants** [comment]
I also get the
sox FAIL formats: can't open output file `Lyrebird-Output': can not open audio device: No such entity

Using archlinux and installed using aur, not sure how to fix that.

**Comment by anashraf002** [comment]
any update?

**Comment by junes** [comment]
How it works for me:

I am using Lyrebird v1.1 on Ubuntu 20.04 installed from the Debian package. 

1. I start Lyrebird from a terminal and see these messages:
```
~$ lyrebird &
Module module-null-sink not loaded.
Module module-remap-source not loaded.
```
2. I manually load the modules:
```
~$ load-module module-null-sink sink_name='Lyrebird-Output'
~$ load-module module-remap-source source_name='Lyrebird-Output'
```
3. I choose a voice pattern by pressing on one of the buttons and then turn the Lyrebird Toggle switch to on.

Everything works as desired.

**Comment by ivanhub** [comment]
Same issue:
"sox FAIL formats: can't open output file `Lyrebird-Output': can not open audio device: No such entity"
I only was able to fix "Module module-null-sink not loaded. \nModule module-remap-source not loaded." issue by the previous post.

**Comment by theRoboxx** [comment]
Did you also install the following packages for Ubuntu?
```
sudo apt-get install sox 
sudo apt-get install libsox-fmt-all 
```

At first I had problems as well but now the sox error is gone.

**But for it to work, I had to load the modules while lyrebird was running**. 
Now I simply opened another two terminals and typed:
(Note that the first command will open pulseaudio with a new output device and therefore you need one terminal per command.  Don't close this pulseaudio-window while lyrebird is open).

```
~$ pavucontrol load-module module-null-sink sink_name='Lyrebird-Output'
~$ pavucontrol load-module module-remap-source source_name='Lyrebird-Output'
```

I tested it with OBS and it works as long as I don't turn off the voice changer button. 
In this case just close lyrebird and repeat the process.

**Comment by ikesaber** [comment]
I'm trying to get lyrebird to work on Ubuntu 22.04.1 LTS (testing using Audacity to record). I've installed all dependencies but when I run `lyrebird` i receive an error:

```
user@computer:~$ lyrebird
Module module-null-sink not loaded.
Module module-remap-source not loaded.
```

I found instances of this issue [34](https://github.com/lyrebird-voice-changer/lyrebird/issues/34) and [96](https://github.com/lyrebird-voice-changer/lyrebird/issues/96) but haven't been able to resolve the issue on my machine. 

I was hoping [these instrucitons to load modules in other terminal windows](https://github.com/lyrebird-voice-changer/lyrebird/issues/34#issuecomment-1004752234) would help but they didn't. My voice just doesn't change...

here's what I ran specifically:

*Window 1*
```
user@computer:~$ lyrebird
Module module-null-sink not loaded.
Module module-remap-source not loaded.
Loaded null output sink (0), and remap sink (0)
```

*New window*
```
user@computer:~$ pavucontrol load-module module-null-sink sink_name='Lyrebird-Output'
```

*New window*
```
user@computer:~$ pavucontrol load-module module-remap-source source_name='Lyrebird-Output'
```

"SoX: playback *on* is set to "Lyrebird Output""

*New window*
```
user@computer:~$ audacity
```

Do you have a fix for this? Am I doing something wrong here?

**Comment by harrego** [comment]
This has been fixed in #111 and will be included in v1.1.1
